packages/loader/container-loader/src/container.ts

```diff
@@ -2112,6 +2117,25 @@ export class Container
 		}
 		const local = this.clientId === message.clientId;
 
+		// Check and report if we're getting messages from a clientId that we previously
+		// flagged should have left, or from a client that's not in the quorum but should be
+		if (message.clientId != null) {
+			const client = this.protocolHandler.quorum.getMember(message.clientId);
+
+			if (client === undefined && message.type !== MessageType.ClientJoin) {
+				// pre-0.58 error message: messageClientIdMissingFromQuorum
+				throw new Error("Remote message's clientId is missing from the quorum");
+			}
+
+			if (
+				this.clientsWhoShouldHaveLeft.has(message.clientId) &&
+				!canBeCoalescedByService(message)
```

Comment:
LGTM! I am curious about what this canBeCoalescedByService checks for though.

Reply:
I believe the original idea is that if the service receives multiple noops from a client in a row, it can coalesce them together (it's not semantically meaningful to receive lots of noops vs. just a single one).  There's some [logic in DeltaManager.processInboundMessage](https://github.com/microsoft/FluidFramework/blob/main/packages/loader/container-loader/src/deltaManager.ts#L956) to handle "missing" noops that were coalesced in this way.

Here it's kind of a weird use though.  In this location it's being used more to test "Although this message came from a client we weren't expecting, is benign like a noop?  If it's benign, we don't have to blow up so aggressively and can let it slide".  So it's not really about being coalesceable in this location, so much as a "isBenignToProcessAnyway" check :-P

Reply:
Added a comment to this effect in latest too.